lib/rubocop/cop/cop.rb

```diff
@@ -183,8 +189,19 @@ def relevant_file?(file)
           !file_name_matches_any?(file, 'Exclude', false)
       end
 
+      def style_guide_url
+        url = cop_config && cop_config['StyleGuide']
+        (url.nil? || url.empty?) ? nil : url
+      end
+
       private
 
+      def annotate_message(message)
+        message = "#{name}: #{message}" if display_cop_names?
+        message = "#{message} (#{style_guide_url})" if display_style_guide?
+        message
+      end
```

Comment:
I’m not in favor of changing the `message` variable – I would prefer something like this:
```ruby
def annotate_message(message)
  if display_cop_names?
    "#{name}: #{message}"
  elsif display_style_guide?
    "#{message} (#{style_guide_url})"
  else
    message
  end
end
```